src/Core/Resolvers/MultipleCreateOrderHelper.cs

```diff
@@ -61,6 +64,26 @@ public class MultipleCreateOrderHelper
                     subStatusCode: DataApiBuilderException.SubStatusCodes.EntityNotFound);
             }
 
+            DatabaseTable sourceDbTable = (DatabaseTable)sourceDbObject;
+            DatabaseTable targetDbTable = (DatabaseTable)targetDbObject;
+            if (sourceDbTable.Equals(targetDbTable))
+            {
+                //  DAB does not yet support multiple-create for self referencing relationships where both the source and
+                //  target entities are backed by same database table.
+                throw new DataApiBuilderException(
+                message: $"Multiple-create for relationship: {relationshipName} at level: {nestingLevel} is not supported because " +
+                $"source entity: {sourceEntityName} and target entity: {targetEntityName} are backed by same database table.",
+                statusCode: HttpStatusCode.BadRequest,
+                subStatusCode: DataApiBuilderException.SubStatusCodes.NotSupported);
+            }
+
+            if (isMNRelationship)
+            {
+                // For M:N relationships,'neither the source nor the target entity act as the referenced entity.
```

Comment:
```suggestion
                // For M:N relationships, neither the source nor the target entity act as the referencing entity.
```

Reply:
why is this change part of this validation PR, it seems like this is a bug fix

Reply:
Not a bug fix. Previously, this method was not even called for M:N relationships. Reasoning added to the PR description.